tests/network_rules_update_test.go

```diff
@@ -0,0 +1,204 @@
+package tests
+
+import (
+	"encoding/json"
+	"github.com/0xsoniclabs/sonic/evmcore"
+	"github.com/0xsoniclabs/sonic/gossip/contract/driverauth100"
+	"github.com/0xsoniclabs/sonic/opera/contracts/driverauth"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/require"
+	"math/big"
+	"testing"
+)
+
+func TestNetworkRule_Update_RulesChangeDuringEpochHasNoEffect(t *testing.T) {
+	require := require.New(t)
+	net := StartIntegrationTestNetWithFakeGenesis(t)
+	defer net.Stop()
```

Comment:
nit: this is no longer needed. The function above is registering a shut-down handler in `t` automatically
```suggestion
```